### DISCUSSION TITLE
More uBlock Options

### DISCUSSION THREAD

**Discussion by ghost** [uBlockOrigin/uBlock-issues#1752] [Ideas]
Some ideas for very useful uBlock options, so we don't have to install seperate plugins to do this.

Useful uBlock Options:
* Clear Site Data
* Block Redirects
  * ``<meta http-equiv="refresh" content="0;URL='http://domain.com/'" />``
* User Agent Spoofer
* Referer Spoofer
* ETag Spoofer
* X-Forwarded-For Spoofer
* Hosts Blocker
* Disable JavaScript
* Disable Flash
* Disable Auto Video Playback
* Cookies Cleaner
* Headers Spoofer (various well known headers with custom strings this can include ETag etc)
* XSS Protection

Just ideas, but some can be extremely useful. Most of this can be off by default of course and an option for advanced mode. But most of this options is very well suated for plugin like uBlock. To me uBlock is more than basic ads blocker, its like a browser firewall. And it should include more options like this for advanced users. Thank you!

**Comment by ghost** [comment]
uBO is not anti-fingerprinting add-on. Many of these are anti-fingerprinting - it's better for it to be handled comprehensively by browser.


    Clear Site Data

Why not just press ctrl+shift_del? What special with it?

    Block Redirects
        <meta http-equiv="refresh" content="0;URL='http://domain.com/'" />

Achievable in Firefox, next version will have scriptlet for Chrome.

    User Agent Spoofer
    Referer Spoofer
    ETag Spoofer
    X-Forwarded-For Spoofer

A FP...

    Hosts Blocker

So... what is uBO already doing?

    Disable JavaScript

Already available by "No-scripting" switch for ex.

    Disable Flash

Already dead.

    Disable Auto Video Playback

Partially achievable by no-large-media.

    Cookies Cleaner

Partially achievable by scriptlet. If you block content, no cookie will be set.

    Headers Spoofer (various well known headers with custom strings this can include ETag etc)

A FP...

    XSS Protection

What exactly? Just block 3p frames/cookies.

**Comment by gorhill** [comment]
It's easy to have ideas, and especially when they are thrown as someone else's burden to work on. I constantly have new ideas, and I even work on them sometimes just to throw them out in the end. The real value is researching and fleshing out ideas, and coming up with prototype to validate them.

I don't want the Discussions to be turned into a place for people to post bulleted laundry list for me to work on, I will quickly close the section if it comes to this. I welcome thoroughly researched ideas, not superficial laundry list which take no more than trivial efforts to post.

**Comment by uBlock-user** [comment]
OP has a history of doing this - https://github.com/uBlockOrigin/uBlock-issues/issues?q=is%3Aissue+author%3Av00v

His most recent -- https://github.com/uBlockOrigin/uBlock-issues/issues/1428
